src/MultiFormats/Registry/HashingAlgorithm.cs

```diff
@@ -0,0 +1,311 @@
+#region LICENSE
+
+/**
+* Copyright (c) 2019 Catalyst Network
+*
+* This file is part of Catalyst.Node <https://github.com/catalyst-network/Catalyst.Node>
+*
+* Catalyst.Node is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Security.Cryptography;
+using MultiFormats.Cryptography;
+using BC = Org.BouncyCastle.Crypto.Digests;
+
+namespace MultiFormats.Registry
+{
+    /// <summary>
+    ///   Metadata and implemetations of a IPFS hashing algorithms.
+    /// </summary>
+    /// <remarks>
+    ///   IPFS assigns a unique <see cref="Name"/> and <see cref="Code"/> to a hashing algorithm. 
+    ///   See <see href="https://github.com/multiformats/multicodec/blob/master/table.csv">hashtable.csv</see>
+    ///   for the currently defined hashing algorithms.
+    ///   <para>
+    ///   These algorithms are implemented:
+    ///   <list type="bullet">
+    ///   <item><description>blake2b-160, blake2b-256 blake2b-384 and blake2b-512</description></item>
+    ///   <item><description>blake2s-128, blake2s-160, blake2s-224 a nd blake2s-256</description></item>
+    ///   <item><description>keccak-224, keccak-256, keccak-384 and keccak-512</description></item>
+    ///   <item><description>md4 and md5</description></item>
+    ///   <item><description>sha1</description></item>
+    ///   <item><description>sha2-256, sha2-512 and dbl-sha2-256</description></item>
+    ///   <item><description>sha3-224, sha3-256, sha3-384 and sha3-512</description></item>
+    ///   <item><description>shake-128 and shake-256</description></item>
+    ///   </list>
+    ///   </para>
+    ///   <para>
+    ///   The <c>identity</c> hash is also implemented;  which just returns the input bytes.
+    ///   This is used to inline a small amount of data into a <see>
+    ///       <cref>Lib.P2P.Cid</cref>
+    ///   </see>
+    ///   .
+    ///   </para>
+    ///   <para>
+    ///     Use <see cref="Register(string, int, int, Func{HashAlgorithm})"/> to add a new
+    ///     hashing algorithm.
+    ///   </para>
+    /// </remarks>
+    public class HashingAlgorithm
+    {
+        internal static Dictionary<string, HashingAlgorithm> Names = new Dictionary<string, HashingAlgorithm>();
+        internal static Dictionary<int, HashingAlgorithm> Codes = new Dictionary<int, HashingAlgorithm>();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Change the visibility of 'Codes' or make it 'const' or 'readonly'.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)